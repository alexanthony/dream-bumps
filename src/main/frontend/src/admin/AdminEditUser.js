// @flow
import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import {Card, CardTitle, CardText} from 'material-ui/Card';
import {Field, reduxForm} from 'redux-form';
import {TextField, SelectField} from 'redux-form-material-ui';
import RaisedButton from 'material-ui/RaisedButton';
import {Cell} from 'material-grid/dist';
import Select from 'react-select';
import MenuItem from 'material-ui/MenuItem';
import 'react-select/dist/react-select.css';

import { getUser } from '../rootReducer';
import { updateUser } from './actions';
import colleges from '../utils/colleges';

const authOptions = [
  {label: 'ROLE_USER', value: 'ROLE_USER'},
  {label: 'ROLE_ADMIN', value: 'ROLE_ADMIN'}
];

const MyReactSelect = (props) => (
  <Select
    {...props}
    value={props.input.value}
    onChange={(value) => props.input.onChange(value.map(val => val.value))}
    onBlur={() => props.input.onBlur(props.input.value)}
    options={props.options}
  />
);

MyReactSelect.propTypes = {
  input: PropTypes.shape({
    value: PropTypes.string,
    onChange: PropTypes.func,
    onBlur: PropTypes.func
  }),
  options: PropTypes.object
}

const AdminEditUser = (
  {handleSubmit, params}: {handleSubmit: Function, params: Object}
) => (
  <Cell col="6" offset={3} >
  <Card>
    <CardTitle title={`Edit User - ${params.userId}`} />
    <CardText>
      <form onSubmit={handleSubmit}>
          <Field
            name="login"
            component={TextField}
            floatingLabelText="Login"
            fullWidth
          />
          <Field
            name="firstName"
            component={TextField}
            floatingLabelText="First Name"
            fullWidth
          />
          <Field
            name="lastName"
            component={TextField}
            floatingLabelText="Last Name"
            fullWidth
          />
          <Field
            name="email"
            component={TextField}
            floatingLabelText="E-Mail"
            fullWidth
          />
          <Field
            name="authorities"
            component={MyReactSelect}
            multi
            options={authOptions}
          />
          <Field
            name="college"
            fullWidth
            floatingLabelText="College"
            component={SelectField}
          >
            {colleges.map(college =>
              <MenuItem
                key={college.value}
                value={college.value}
                primaryText={college.label}
              />
            )}
          </Field>
        <RaisedButton primary type="submit" label="Save" />
      </form>
    </CardText>
  </Card>
  </Cell>
);

AdminEditUser.propTypes = {
  handleSubmit: PropTypes.func,
  params: PropTypes.shape({
    userId: PropTypes.string
  }),
  initialValues: PropTypes.shape({
    login: PropTypes.string,
    firstName: PropTypes.string,
    lastName: PropTypes.string,
    email: PropTypes.string,
    authorities: PropTypes.arrayOf(PropTypes.string),
    college: PropTypes.string
  })
}

// todo - activated and authorities

export const mapStateToProps = (state: Object, {params}:{params:Object}) => ({
  initialValues: getUser(state)(params.userId)
});
export {AdminEditUser};

export default connect(
  mapStateToProps,
  {onSubmit: updateUser}
)(
  reduxForm(
    {form: 'edit-user', enableReinitialize: true}
  )(AdminEditUser));
