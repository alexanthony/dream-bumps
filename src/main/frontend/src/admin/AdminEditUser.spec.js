import React from 'react';
import { shallow } from 'enzyme';
import { AdminEditUser } from './AdminEditUser';
import toJson from 'enzyme-to-json';

test('Renders properly', () => {
  const submit = jest.fn();
  const component = shallow(
    <AdminEditUser
      params={{userId: '21'}}
      handleSubmit={submit}
      initialValues={{id: 21, login: 'username', firstName: 'Joe', lastName: 'Bloggs', email: 'joe@bloggs.com', authorities: ['ROLE_USER'], college: null}}
    />
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot()
});

// AdminEditUser.propTypes = {
//   user: PropTypes.object,
//   handleSubmit: PropTypes.func,
//   params: PropTypes.object,
//   initialValues: PropTypes.object
// }
