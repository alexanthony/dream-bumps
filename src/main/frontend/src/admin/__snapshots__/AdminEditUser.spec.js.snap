exports[`test Renders properly 1`] = `
<Cell
  col="6"
  offset={3}>
  <Card
    expandable={false}
    expanded={null}
    initiallyExpanded={false}>
    <CardTitle
      title="Edit User - 21" />
    <CardText>
      <form
        onSubmit={[Function]}>
        <Field
          component={[Function]}
          floatingLabelText="Login"
          fullWidth={true}
          name="login" />
        <Field
          component={[Function]}
          floatingLabelText="First Name"
          fullWidth={true}
          name="firstName" />
        <Field
          component={[Function]}
          floatingLabelText="Last Name"
          fullWidth={true}
          name="lastName" />
        <Field
          component={[Function]}
          floatingLabelText="E-Mail"
          fullWidth={true}
          name="email" />
        <Field
          component={[Function]}
          multi={true}
          name="authorities"
          options={
            Array [
              Object {
                "label": "ROLE_USER",
                "value": "ROLE_USER",
              },
              Object {
                "label": "ROLE_ADMIN",
                "value": "ROLE_ADMIN",
              },
            ]
          } />
        <Field
          component={[Function]}
          floatingLabelText="College"
          fullWidth={true}
          name="college">
          <MenuItem
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText=""
            value={null} />
          <MenuItem
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="New College"
            value="newc" />
          <MenuItem
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="Balliol"
            value="balliol" />
        </Field>
        <RaisedButton
          disabled={false}
          fullWidth={false}
          label="Save"
          labelPosition="after"
          primary={true}
          secondary={false}
          type="submit" />
      </form>
    </CardText>
  </Card>
</Cell>
`;
